src/main/java/org/springframework/cloud/deployer/spi/kubernetes/AbstractKubernetesDeployer.java

```diff
@@ -127,14 +129,29 @@ protected AppStatus buildAppStatus(String id, PodList podList, ServiceList servi
 		Service service = null;
 		if (podList != null && podList.getItems() != null) {
 			for (Pod pod : podList.getItems()) {
+				String deploymentKey = pod.getMetadata().getLabels().get(SPRING_DEPLOYMENT_KEY);
 				for (Service svc : services.getItems()) {
-					if (svc.getMetadata().getName()
-							.equals(pod.getMetadata().getLabels().get(SPRING_DEPLOYMENT_KEY))) {
+					if (svc.getMetadata().getName().equals(deploymentKey)) {
 						service = svc;
 						break;
 					}
 				}
-				statusBuilder.with(new KubernetesAppInstanceStatus(pod, service, properties));
+				//find the container with the correct env var
+				for(Container container : pod.getSpec().getContainers()) {
+					if(container.getEnv().stream().anyMatch(envVar -> "SPRING_CLOUD_APPLICATION_GROUP".equals(envVar.getName()))) {
```

Comment:
i think using `SPRING_CLOUD_APPLICATION_GUID` would be better here as its not conditionally added based on a deployment property as with `SPRING_CLOUD_APPLICATION_GROUP`.

Reply:
Thanks!! I will add that right now.